--- a/repositories/stable-diffusion-stability-ai/ldm/models/diffusion/ddpm.py	2024-04-01 05:10:11.038199192 +0200
+++ b/repositories/stable-diffusion-stability-ai/ldm/models/diffusion/ddpm.py	2024-03-30 22:58:56.472609011 +0100
@@ -17,7 +17,7 @@
 import itertools
 from tqdm import tqdm
 from torchvision.utils import make_grid
-from pytorch_lightning.utilities.distributed import rank_zero_only
+from pytorch_lightning.utilities.rank_zero import rank_zero_only
 from omegaconf import ListConfig
 
 from ldm.util import log_txt_as_img, exists, default, ismap, isimage, mean_flat, count_params, instantiate_from_config
