--- a/modules/paths_internal.py	2024-04-01 21:02:27.482198100 +0200
+++ b/modules/paths_internal.py	2024-04-01 21:02:25.192211115 +0200
@@ -31,7 +31,7 @@
 models_path = os.path.join(data_path, "models")
 extensions_dir = os.path.join(data_path, "extensions")
 extensions_builtin_dir = os.path.join(script_path, "extensions-builtin")
-config_states_dir = os.path.join(script_path, "config_states")
+config_states_dir = os.path.join(data_path, "config_states")
 default_output_dir = os.path.join(data_path, "outputs")
 
 roboto_ttf_file = os.path.join(modules_path, 'Roboto-Regular.ttf')
